File: 01585.py
n = int(input())

while n > 0:
    s = str(input())
    ans = 0
    temp = 0

    for char in s:
        if char == "O":
            temp += 1
        else:
            temp = 0
        ans += temp

    print(ans)
    n -= 1

File: 01585.cpp
#include <iostream>
using namespace std;
#include <string>
int main()
{
    int n;
    cin>>n;
    while (n--)
    {
        string s;
        cin >> s;
        int ans = 0;
        int temp = 0;
        for (int i = 0; i < s.length(); i++)
        {
            if (s[i] == 'O')
            {
                temp++;
            }
            else if (s[i] == 'X')
            {
                temp = 0;
            }
            ans += temp;
        }
        cout << ans << endl;
    }
}
